.github/workflows/publish.yml

```diff
@@ -27,6 +27,10 @@ jobs:
       - uses: actions/checkout@v4
       - uses: docker/setup-qemu-action@v3
       - uses: docker/setup-buildx-action@v3
+      - uses: docker/login-action@v3 # helps with rate limit issues
```

Comment:
Shouldn't this happen as the first docker step? To make sure any images that the previous steps pull are pulled while logged in? e.g. `docker/setup-qemu-action` pulls the `binfmt` image afaik.